app/src/main/java/org/gdg/frisbee/android/common/GdgActivity.java

```diff
@@ -118,69 +108,86 @@ public GoogleApiClient getGoogleApiClient() {
+        }
+
         mGoogleApiClient = createGoogleApiClient();
 
         mAchievementActionHandler =
             new AchievementActionHandler(mGoogleApiClient, this);
     }
 
-    protected GoogleApiClient createGoogleApiClient() {
-        GoogleApiClient.Builder builder = new GoogleApiClient.Builder(this)
-            .addApi(AppIndex.API);
-
-        if (PrefUtils.isSignedIn(this)) {
-            builder.addApi(Plus.API).addScope(Plus.SCOPE_PLUS_LOGIN).addScope(Plus.SCOPE_PLUS_PROFILE)
-                .addApi(Games.API).addScope(Games.SCOPE_GAMES)
-                .addApi(Drive.API).addScope(Drive.SCOPE_APPFOLDER);
-        }
-
-        return builder.build();
-    }
-
     @Override
     protected void onStart() {
         super.onStart();
 
-        mGoogleApiClient.registerConnectionCallbacks(this);
-        mGoogleApiClient.registerConnectionFailedListener(this);
-
+        recreateGoogleApiClientIfNeeded();
         mGoogleApiClient.connect();
     }
 
+    protected final void recreateGoogleApiClientIfNeeded() {
+        if (isSignedIn != PrefUtils.isSignedIn(this)) {
+            mGoogleApiClient.unregisterConnectionCallbacks(this);
+            mGoogleApiClient.unregisterConnectionFailedListener(this);
+            mGoogleApiClient.disconnect();
+
+            mGoogleApiClient = createGoogleApiClient();
+        }
+
+        if (!mGoogleApiClient.isConnectionCallbacksRegistered(this)) {
+            mGoogleApiClient.registerConnectionCallbacks(this);
+        }
+        if (!mGoogleApiClient.isConnectionFailedListenerRegistered(this)) {
+            mGoogleApiClient.registerConnectionFailedListener(this);
+        }
+    }
+
+    /**
+     * Create {@link GoogleApiClient}. This can be overridden to change the scope of the GoogleApiClient
+     *
+     * @return {@link GoogleApiClient} without connecting. {@code connect()} must be called afterwards.
+     */
+    protected GoogleApiClient createGoogleApiClient() {
+        isSignedIn = PrefUtils.isSignedIn(this);
+        return GoogleApiClientFactory.createWith(getApplicationContext());
+    }
+
     @Override
     protected void onStop() {
         super.onStop();
 
         mGoogleApiClient.unregisterConnectionCallbacks(this);
         mGoogleApiClient.unregisterConnectionFailedListener(this);
-
-        if (mGoogleApiClient.isConnected()) {
-            mGoogleApiClient.disconnect();
-        }
+        mGoogleApiClient.disconnect();
     }
 
-    public AchievementActionHandler getAchievementActionHandler() {
-        return mAchievementActionHandler;
+    @Override
+    protected void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        outState.putInt(SAVED_PROGRESS, mSignInProgress);
     }
 
     @Override
     protected void onActivityResult(int requestCode, int responseCode, Intent intent) {
         super.onActivityResult(requestCode, responseCode, intent);
 
-        if (PrefUtils.isSignedIn(this) && !Utils.isEmulator()) {
```

Comment:
This was here since Emulators had problems with Google Play Services.

Right now, we are trying to make the app work even there is no GMS or user does not have Google+ Google Play Games account. The app will work on every situation. So there is no need for emulator check.